### PR TITLE
Allow JavaScriptAgent to get and set user credentials

### LINKED ISSUE #1022: Allow JavaScript agents to update Credentials
Allow JavaScript agents to update Credentials, making Credentials a kind of shared memory.
Maybe something like:

    this.credentials()
    this.credentials(name)
    this.credentials(nameToSet, valueToSet)

In this way the Credentials may be updated by an Agent and accessed be another Agent.
This could also help resolving issues like #1009

### PR DESCRIPTION
**Pull request by dsander**
Fixes #1022

### CODE REVIEW

File: spec/models/agents/java_script_agent_spec.rb [hunk 1]
```diff
@@ -264,5 +264,33 @@
         expect(AgentLog.last.message).to eq("hello from coffeescript")
       end
     end
+
+    describe "user credentials", focus: true do
```
Comment (cantino) [inline spec/models/agents/java_script_agent_spec.rb]: Should remove `focus: true`, right?

**Comment by cantino** [comment]
Looks perfect, thanks @dsander!

File: spec/models/agents/java_script_agent_spec.rb (hunk 1 above)
Comment (dsander) [inline spec/models/agents/java_script_agent_spec.rb]: Woups, thanks.

**Comment by dsander** [comment]
Thanks, merging.

**Pull request merged**